core/blockchain.go

```diff
@@ -2082,6 +2091,9 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 		if err != nil {
 			return it.index, err
 		}
+
+		bc.cacheReceipts(block.Hash(), receipts, block)
```

Comment:
Why this line's positioning was changed?

Reply:
no need to store the field `EffectiveGasPrice` to disk
because it can be derived from transaction info, please refer to `func ReadReceipts`-->`func DeriveFields`
so we move the cache action after `func WriteReceipts`